solutions/usaco-916.mdx

```diff
@@ -66,6 +66,40 @@ public class Revegetation {
 ```
 
 </JavaSection>
+
+<PySection>
+
+```python
+MAX_PASTURE = 151
+	
+a, b, g = [0] * MAX_PASTURE, [0] * MAX_PASTURE, [0] * MAX_PASTURE
```

Comment:
wait i took a closer look at your code
why can't you just dynamically size these lists? they're dependent on input size, no?

Reply:
yes but technically, isnt this faster?

Reply:
how?

Reply:
like appending in [] is slow

Reply:
can i have an SO link or something for that?

Reply:
u know how arraylists work right? it just creates a new array of 1 greater size every time u append, which is why people use linked list to speed up (when u are appending or deleting a lot) but if u need to get the index u use a array because you can access the index in O(1). Thats basically python list (the arraylist) so, its slower if the "computer" keeps creating a new static array everytime u append.

Reply:
ill try to find the link rn

Reply:
https://www.geeksforgeeks.org/internal-working-of-list-in-python/#:~:text=Python%20lists%20are%20internally%20represented,if%20preallocated%20space%20becomes%20full.

Reply:
```py
In [1]: def rep_append(sz: int) -> [int]:
   ...:     res = []
   ...:     for i in range(sz):
   ...:         res.append(0)
   ...:     return res
   ...:

In [2]: def all_at_once(sz: int) -> [int]:
   ...:     return [0] * sz
   ...:

In [3]: %timeit rep_append(10000)
357 µs ± 1.64 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [4]: %timeit all_at_once(10000)
22 µs ± 61.4 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```
hm, you seem to be right
thing is, i'm not suggesting repeated appending
i'm suggesting more of just sizing the list based on input
instead of
```py
a = [0] * MAX_PASTURE
```
why not just do this instead:
```py
a = [0] * m
```

Reply:
and even if repeated appending was slower, as you can see it still really has a negligible impact on the runtime

Reply:
it doesnt work. idk why
<img width="725" alt="Screen Shot 2022-05-20 at 12 08 54 AM" src="https://user-images.githubusercontent.com/52589420/169347179-ecff1741-2c7a-4fb4-9de6-626260b5752b.png">

Reply:
download the test cases?

Reply:
isdk why it doesn't work

Reply:
download the cases then?

Reply:
how hard can it be to debug a lil

Reply:
okay ill debug but whats the point... using staticsized arrays work fine... and technically its even faster

Reply:
speed doesn't really matter as long as it's AC

Reply:
yeah but whats wrong with using staticsized arrays

Reply:
dynamic >> static
handles more cases and stuff

Reply:
what does bump mean @SansPapyrus683

Reply:
changed and it works
<img width="852" alt="Screen Shot 2022-07-22 at 12 08 56 AM" src="https://user-images.githubusercontent.com/52589420/180261884-29ffa72f-918f-468f-ad6d-49d56886c2b8.png">
<img width="232" alt="Screen Shot 2022-07-22 at 12 09 26 AM" src="https://user-images.githubusercontent.com/52589420/180261969-6f086ffb-9d6f-4050-ab1d-60a9aad0cd3c.png">

Reply:
nice